### PR TITLE
DeepSeek-V3/R1 MoE load balance deployment and inference using EPLB

### PR DESCRIPTION
**Pull request by feliang-git**
## Motivation

To achieve load balance during inference, a crucial feature is to load expert by a given combination, instead of loading them sequentially. Given that the load traffic monitor tool and DeepSeek EPLB algorithm are ready, this functionality would be the next step in achieving MoE expert load balance.

## Modifications

 - Modified EPMoE class weight_loading and forward methods to support expert mapping.
 - Modified DeepseekV2MoE class. Passing MoE expert mapping tensor.
 - Added a server start option `--enable-eplb-moe` to turn on MoE expert load balance.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/epmoe_permute_tensor.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+torch.manual_seed(42)
+
+# Create a global tensor variable for testing
+# 61 rows, each row has a random permutation of integers from 0 to 255
+EP_PERMUTE_TENSOR = torch.stack([
+    torch.randperm(256)
+    for _ in range(61)
+], dim=0)
+
+EP_BACK_MAPPING_TENSOR = torch.zeros((61, 256), dtype=torch.long)
```
Comment (liz-badada) [inline python/sglang/srt/epmoe_permute_tensor.py:11]: Better pass the num_layers info as well as num_experts.

File: python/sglang/srt/models/deepseek_v2.py [hunk 2]
```diff
@@ -291,6 +293,7 @@ def forward_normal(self, hidden_states: torch.Tensor) -> torch.Tensor:
             final_hidden_states = final_hidden_states + shared_output
         if self.tp_size > 1:
             final_hidden_states = tensor_model_parallel_all_reduce(final_hidden_states)
+        print(f"final_hidden_states: {final_hidden_states}")
```
Comment (liz-badada) [inline python/sglang/srt/models/deepseek_v2.py]: debugging code?

File: python/sglang/srt/managers/schedule_batch.py [hunk 3]
```diff
@@ -55,6 +55,7 @@
 from sglang.srt.sampling.sampling_params import SamplingParams
 from sglang.srt.server_args import ServerArgs
 from sglang.srt.utils import flatten_nested_list, get_compiler_backend
+from sglang.srt.epmoe_permute_tensor import EP_PERMUTE_TENSOR, EP_BACK_MAPPING_TENSOR
```
Comment (liz-badada) [inline python/sglang/srt/managers/schedule_batch.py]: for EP_PERMUTE_TENSOR and EP_BACK_MAPPING_TENSOR, are these two global tensors would be replaced by the output of [EPLB](https://github.com/deepseek-ai/EPLB) in the future? Currently they are just random generated.

Comment (feliang-git) [reply python/sglang/srt/managers/schedule_batch.py]: Yes, these two tensors are just a placeholder for EPLB results.

File: python/sglang/srt/models/deepseek_v2.py (hunk 2 above)
Comment (feliang-git) [reply python/sglang/srt/models/deepseek_v2.py]: Removed!

**Comment by jokerwyt** [comment]
Hi, any progress on this PR? Does it work on `--enable-ep-moe` or `--enable-deepep-moe`? @feliang-git

**Comment by feliang-git** [comment]
> Hi, any progress on this PR? Does it work on `--enable-ep-moe` or `--enable-deepep-moe`? @feliang-git

**Comment by feliang-git** [comment]
> Hi, any progress on this PR? Does it work on `--enable-ep-moe` or `--enable-deepep-moe`? @feliang-git

Hi @jokerwyt, there is another pr working on implementing EPLB. Please refer to [pr#5295](https://github.com/sgl-project/sglang/pull/5295) for more detail. Thanks!
